lib/virtus/support/type_lookup.rb

```diff
@@ -84,7 +82,7 @@ def determine_type_from_primitive(primitive)
       descendants.select(&:primitive).reverse_each do |descendant|
         descendant_primitive = descendant.primitive
         next unless primitive <= descendant_primitive
-        type = descendant if type.nil? or type.primitive > descendant_primitive
+        type = descendant if type.nil? || type.primitive > descendant_primitive
```

Comment:
@envygeeks That's incorrect, `||` has a higher precedence than `or`. 

`a = false or true` sets `a` to `false`, and `a = false || true` sets `a` to `true`, since = precedence is lower than || but higher than or. [SO](http://stackoverflow.com/questions/2083112/difference-between-or-and-in-ruby)